### PR TITLE
UniversalResults: add display: flex to the icon component

### PR DESCRIPTION
**Pull request by oshi97**
This was done in the old resultssectionheader.hbs, and fixes
vertical alignment issues. Adding the original class="yxt-Results-titleIconWrapper"
adds this.

TEST=manual
test that if I set &-titleIconWrapper to an arbitrary height the
icon will be vertically aligned with display: flex and will not be
aligned without it

### CODE REVIEW

File: src/ui/sass/modules/_Results.scss [hunk 1]
```diff
@@ -161,11 +161,6 @@ $results-cards-margin: calc(var(--yxt-base-spacing) / 2) !default;
     background-color:  var(--yxt-color-background-highlight);
   }
 
-  &-title .Icon
-  {
-    margin-right: calc(var(--yxt-base-spacing) / 2);
```
Comment (creotutar) [inline src/ui/sass/modules/_Results.scss:166]: is this margin change intentional? if so, can we update the commit message and/or desc? doesn't seem related to the vertical alignment (unless I'm missing something)

Comment (oshi97) [reply src/ui/sass/modules/_Results.scss:166]: this margin is already added in yxt-Results-titleIconWrapper so is redundant

**Review by creotutar — APPROVED** [review]
🌻 🌻 🌻 LGTM
